src/main/org/firebirdsql/jdbc/InternalTransactionCoordinator.java

```diff
@@ -175,7 +175,7 @@ void handleConnectionClose() throws SQLException {
         coordinator.handleConnectionClose();
     }
 
-    private void setCoordinator(AbstractTransactionCoordinator coordinator) throws SQLException {
+    public void setCoordinator(AbstractTransactionCoordinator coordinator) throws SQLException {
```

Comment:
The default for normal connections is `autoCommit=true`, but this is initialized by calling `setManagedEnvironment(false)` from `FBManagedConnection.getConnection`. Your implementation has overridden that with an empty implementation.

I think it is preferable to leave `setCoordinator` and `LocalTransactionCoordinator` package private, and instead add to FBConnection:

```
protected void setTransactionCoordinator(boolean managedConnection, boolean autoCommit) throws SQLException {
    checkValidity();
    txCoordinator.setTransactionCoordinator(managedConnection, autoCommit);
}
```

and replace `InternalFBConnection.setManagedEnvironment` with

```
@Override
public void setManagedEnvironment(boolean managedConnection)
{
    checkValidity();
    setTransactionCoordinator(managedEnvironment, false);
}
```

I will need to study on this a bit more though.